framework/web/Response.php

```diff
@@ -880,6 +880,10 @@ protected function getDispositionHeaderValue($disposition, $attachmentName)
      */
     public function redirect($url, $statusCode = 302, $checkAjax = true)
     {
+        if(is_null($url)){
+            $url = '/';
+        }
```

Comment:
```suggestion
        if ($url === null) {
            $url = '';
        }
```

Reply:
when $url is null and set empty string,this while be redirect the current link again and again

Reply:
Hmm, right, we need to cast string on $url in both strncmp functions in that case to preserve default behavior.

Reply:
`$url` can't be `null`
~~~php
if ($url == '') {
    $url = '/';
}
~~~